### PR TITLE
Code refactoring, addition of new unit tests and bug fixes

### PR DESCRIPTION
**Pull request by alefiury**
This PR tries to do code refactoring, so that the code becomes more readeable in the long term, add unit tests to the scanner and fixes some scanner bugs. The code refactoring focused more on how the dft is constructed and represented.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,5 +21,6 @@ This is a compiler for mgol implemented in go for the final work of the compiler
 - [x] Check why the readme is a binary file
 - [x] Implement symbol table
 - [x] Implement symbol table tests
-- [ ] Return elements from the symbol table
+- [x] Return elements from the symbol table
 - [ ] Fix the rows and columns bug
+- [ ] Fix bug where the transition does not exist and is not a final state
```
Comment (MatheusNtg) [inline README.md]: If those bugs are already solved, we can marked it as solved.

File: src/lexer/scanner_test.go [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func fillSymbolTable(table *SymbolTable) {
+	for _, languageToken := range LanguageReservedTokens {
+		table.Insert(languageToken.GetLexem(), languageToken)
+	}
+}
```
Comment (MatheusNtg) [inline src/lexer/scanner_test.go]: I was thinking, we use this function both in `main.go` and here, so maybe it should be good put it on `utils.go`, what do you think @alefiury?

**Review by MatheusNtg — CHANGES_REQUESTED** [review]
Overall looks good to me, just small changes required

File: src/lexer/scanner_test.go (hunk 2 above)
Comment (alefiury) [reply src/lexer/scanner_test.go]: I think so too. I am going to add this function to the `utils.go` module.

File: README.md (hunk 1 above)
Comment (alefiury) [reply README.md]: Agreed.

Comment (alefiury) [reply README.md]: Change done.

File: src/lexer/scanner_test.go (hunk 2 above)
Comment (alefiury) [reply src/lexer/scanner_test.go]: Changes done.

**Review by MatheusNtg — APPROVED** [review]

**Pull request merged**